test/unit/picks-component.spec.ts

```diff
@@ -0,0 +1,137 @@
+      options.power = 20
+      dbQueryMock.mockResolvedValueOnce({ rows: [{ counts: 1000 }] })
+      result = await picksComponent.getPickStats(itemId, options)
+    })
+
+    it("should query the favorites that were done by users with power greater and equal than the given power", () => {
+      expect(dbQueryMock).toBeCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining(`voting.power >= `),
+          values: expect.arrayContaining([options.power]),
+        })
+      )
+    })
+
+    it("should return the amount of favorites", () => {
+      expect(result).toEqual({ counts: 1000 })
+    })
+  })
+
+  describe("and the power parameter is not set", () => {
+    beforeEach(async () => {
+      options.power = undefined
+      dbQueryMock.mockResolvedValueOnce({ rows: [{ counts: 1000 }] })
+      result = await picksComponent.getPickStats(itemId, options)
+    })
+
+    it("should query the favorites that were done by users with power greater and equal than default power", () => {
+      expect(dbQueryMock).toBeCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining(`voting.power >= `),
+          values: expect.arrayContaining([1]),
+        })
+      )
+    })
+
+    it("should return the amount of favorites", () => {
+      expect(result).toEqual({ counts: 1000 })
+    })
+  })
+
+  describe("and the user address parameter is set", () => {
+    beforeEach(async () => {
+      options.userAddress = "aUserAddress"
+      dbQueryMock.mockResolvedValueOnce({ rows: [{ counts: 1000, likedByUser: true }] })
+      result = await picksComponent.getPickStats(itemId, options)
+    })
+
+    it("should not check in the query if the user has liked the item", () => {
+      expect(dbQueryMock).toBeCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining(", (hasLiked.counter > 0) likedByUser"),
+        })
+      )
+      expect(dbQueryMock).toBeCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining("SELECT COUNT(*) counter FROM favorites.picks"),
+          values: expect.arrayContaining([options.userAddress, itemId]),
+        })
+      )
+      expect(dbQueryMock).toBeCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining("GROUP BY (hasLiked.counter)"),
+          values: expect.arrayContaining([options.userAddress, itemId]),
+        })
+      )
+    })
+
+    it("should return the amount of favorites and the liked by user property", () => {
+      expect(result).toEqual({ counts: 1000, likedByUser: true })
+    })
+  })
+
+  describe("and the user address parameter is not set", () => {
+    beforeEach(async () => {
+      options.userAddress = undefined
+      dbQueryMock.mockResolvedValueOnce({ rows: [{ counts: 1000 }] })
+      result = await picksComponent.getPickStats(itemId, options)
+    })
+
+    it("should check in the query if the user has liked the item", () => {
```

Comment:
and here is not i believe
```suggestion
    it("should not check in the query if the user has liked the item", () => {
```

Reply:
Changed it as well.